### DISCUSSION TITLE
Improve samples documentation

### DISCUSSION THREAD

**Discussion by richbai90** [zephyrproject-rtos/zephyr#38481] [Ideas]
**Is your feature request related to a problem? Please describe.**
The samples don't include many comments explaining what each line is doing. As someone coming from a hobbyist arduino background, trying to learn the ins and outs of true embedded development, it would be extremely useful to have comments explaining what is going on.

**Describe the solution you'd like**
The samples make ample use of macros, which are particularly hard to follow. A simple line of explanation on a line like this `const struct device *mic_dev = device_get_binding(DT_LABEL(DT_INST(0, st_mpxxdtyy))); ` something to the effect of get the device binding of the microphone. DT_INST is doing x, DT_LABEL is doing y, together this expands to z. 

**Describe alternatives you've considered**
Reading the documentation. This is important, no doubt, but better code comments will help beginners.

**Comment by stephanosio** [comment]
Comments are not there to explain the obvious.

The documentation for `device_get_binding`, `DT_LABEL` and `DT_INST` all exist to explain what you describe here.

Of course, we could have a tutorial/special sample that has very detailed comments for the beginners, but that is a whole another issue.
